daemon/container_operations_unix.go

```diff
@@ -760,10 +764,6 @@ func (daemon *Daemon) connectToNetwork(container *container.Container, idOrName
 	controller := daemon.netController
 
 	if endpointConfig != nil {
```

Comment:
Since you have reintroduced `endpointConfig` as a parameter in `updateNetworkConfig` function, can you please move the entire `if` block ? They all face the same issue.